components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/SchemaValidator.java

```diff
@@ -0,0 +1,610 @@
+                messageContext).getAxis2MessageContext();
+        String contentType;
+        Object objContentType = axis2MC.getProperty(ThreatProtectorConstants.REST_CONTENT_TYPE);
+        if (objContentType != null) {
+            contentType = objContentType.toString();
+            if (logger.isDebugEnabled()) {
+                logger.info("Content type of the request message :" + contentType);
+            }
+            try {
+                RelayUtils.buildMessage(axis2MC);
+                if (logger.isDebugEnabled()) {
+                    logger.info("Successfully built the request message");
+                }
+                if (ThreatProtectorConstants.APPLICATION_JSON.equals(contentType)) {
+                    try {
+                        initialize(messageContext);
+                    } catch (APIManagementException e) {
+                        logger.error("Error occurred while initializing the swagger elements", e);
+                    }
+                    if (swagger != null) {
+                        requestMethod = messageContext.getProperty(APIMgtGatewayConstants.
+                                ELECTED_REQUEST_METHOD).toString();
+                        if (!APIConstants.SupportedHTTPVerbs.GET.name().equals(requestMethod)) {
+                            try {
+                                validateRequest(messageContext);
+                            } catch (APIManagementException e) {
+                                logger.error("Error occurred while validating the API request", e);
+                            }
+                        }
+                    }
+                }
+            } catch (IOException | XMLStreamException e) {
+                logger.error("Error occurred while building the API request", e);
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Validate the response message.
+     *
+     * @param messageContext Context of the API response
+     * @return Continue true the handler chain
+     */
+    @Override
+    public boolean handleResponse(MessageContext messageContext) {
+        if (logger.isDebugEnabled()) {
+            logger.info("Validating the API response  Body content..");
+        }
+        org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext) messageContext).
+                getAxis2MessageContext();
+        try {
+            RelayUtils.buildMessage(axis2MC);
+            if (logger.isDebugEnabled()) {
+                logger.info("Successfully built the response message");
+            }
+        } catch (IOException e) {
+            logger.error("Error occurred while building the API response", e);
+        } catch (XMLStreamException e) {
+            logger.error("Error occurred while validating the API response", e);
+        }
+        Object objectResponse = axis2MC.getProperty(ThreatProtectorConstants.REST_CONTENT_TYPE);
+        if (objectResponse != null) {
```

Comment:
Same as `handleRequest`. Can't continue when `objectResponse == null`. We can switch the if block